src/systems/filecoin_files/data_transfer/_index.md

```diff
@@ -3,4 +3,92 @@ menuTitle: Data Transfer
+- **Pull Request**: A request to have the other party send data
+- **Requestor**: The party that initiates the data transfer request (whether Push or Pull)
+- **Responder**: The party that receives the data transfer request
+- **Data Transfer Voucher**: A wrapper around storage or retrieval data that can identify and validate the transfer request to the other party
+- **Request Validator**: The data transfer module only initiates a transfer when the responder can validate that the request is tied directly to either an existing storage deal or retrieval deal. Validation is not performed by the data transfer module itself. Instead, a request validator inspects the data transfer voucher to determine whether to respond to the request.
+- **Scheduler**:  Once a request is negotiated and validated, actual transfer is managed by a scheduler on both sides. The scheduler is part of the data transfer module but is isolated from the negotiation process. It has access to an underlying verifiable transport protocol and uses it to send data and track progress.
+- **Subscriber**: An external component that monitors progress of a data transfer by subscribing to data transfer events, such as progress or completion.
+- **GraphSync**: The default underlying transfer protocol used by the Scheduler. The full graphsync specification can be found at [https://github.com/ipld/specs/blob/master/block-layer/graphsync/graphsync.md](https://github.com/ipld/specs/blob/master/block-layer/graphsync/graphsync.md)
+
+# Request Phases
+
+There are two basic phases to any data transfer:
+
+1. Negotiation - the requestor and responder agree to the transfer by validating with the data transfer voucher
+2. Transfer - Once both parties have negotiated and agreed upon, the data is actually transferred. The default protocol used to do the transfer is Graphsync
+
+Note that the Negotiation and Transfer stages can occur in seperate round trips,
+or potentially the same round trip, where the requesting party implicitly agrees by sending the request, and the responding party can agree and immediately send or receive data.
+
+# Example Flows
+
+## Push Flow
+
+{{< diagram src="./docs/systems/filecoin_files/data_transfer/push-flow.mmd.svg" title="Data Transfer - Push Flow" >}}
+
+1. A requestor initiates a Push transfer when it wants to send data to another party. 
+2. The requestors' data transfer module will send a push request to the responder along with the data transfer voucher. It also puts the data transfer in the scheduler queue, meaning it expects the responder to initiate a transfer once the request is verified
+3. The responder's data transfer module validates the data transfer request via the Validator provided as a dependency by the responder
+4. The responder's data transfer module schedules the transfer
+5. The responder makes a GraphSync request for the data
+6. The requestor receives the graphsync request, verifies it's in the scheduler and begins sending data
```

Comment:
is "graphsync request" a data structure or function call? if so may be good to have `graphsync.Request`. (potentially defined in `libraries/ipfs/graphsync`. 🤷‍♂